Addons/testExoticQuadrature.hpp

```diff
@@ -0,0 +1,162 @@
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions
+ *    and the following disclaimer in the documentation and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse
+ *    or promote products derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ * INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+ * OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA,
+ * OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * UT-BATTELLE, LLC AND THE UNITED STATES GOVERNMENT MAKE NO REPRESENTATIONS AND DISCLAIM ALL WARRANTIES, BOTH EXPRESSED AND
+ * IMPLIED. THERE ARE NO EXPRESS OR IMPLIED WARRANTIES OF MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE, OR THAT THE USE
+ * OF THE SOFTWARE WILL NOT INFRINGE ANY PATENT, COPYRIGHT, TRADEMARK, OR OTHER PROPRIETARY RIGHTS, OR THAT THE SOFTWARE WILL
+ * ACCOMPLISH THE INTENDED RESULTS OR THAT THE SOFTWARE OR ITS USE WILL NOT RESULT IN INJURY OR DAMAGE. THE USER ASSUMES
+ * RESPONSIBILITY FOR ALL LIABILITIES, PENALTIES, FINES, CLAIMS, CAUSES OF ACTION, AND COSTS AND EXPENSES, CAUSED BY, RESULTING
+ * FROM OR ARISING OUT OF, IN WHOLE OR IN PART THE USE, STORAGE OR DISPOSAL OF THE SOFTWARE.
+ */
+
+
+#include "TasmanianAddons.hpp"
+#include "tasgridCLICommon.hpp"
+#include <cstring>
+
+// Test the output of getRoots().
+inline bool testRootSizes() {
+    bool passed = true;
+    std::vector<std::vector<double>> roots;
+    int nref = 5;
+    std::vector<double> ref_weights(nref), ref_points(nref);
+    TasGrid::OneDimensionalNodes::getGaussLegendre(nref, ref_weights, ref_points);
+    // n = 0, 1, 5
+    std::vector<int> n_vec = {1, 2, 5};
+    for (auto n : n_vec) {
+        roots = TasGrid::getRoots(0, ref_weights, ref_points);
+        if (roots.size() != 0) {
+            std::cout << "ERROR: Test failed in testRootSizes() for n = " << n << std::endl;
+            passed = false;
+        }
+    }
+    return passed;
+}
+
+// Test the attributes of the output of getExoticQuadrature().
+inline bool testBasicAttributes() {
+    bool passed = true;
+    const int n = 3;
+    const int nref = 11;
+    const char *descr = "TEST_DESCRIPTION";
+    auto sinc = [](double x) -> double {return (x == 0.0 ? 1.0 : sin(x) / (x));};
```

Comment:
Just a general note: This is not the correct way to implement the `sinc()` function.

The problem with the `0.0` in the denominator is not just at `x = 0.0` but also for values close to zero. The correct way is to pick a small range around `0.0` and then use a truncated Maclaurin expansion. There's no need to use that here since the Gauss-Legendre quadrature points will fall away from `0.0`.

Reply:
Noted.